x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_overview_logic.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+
+const POLLING_DURATION = 5000;
+
+interface EngineOverviewApiData {
+  apiLogsUnavailable: boolean;
+  documentCount: number;
+  startDate: string;
+  endDate: string;
+  operationsPerDay: number[];
+  queriesPerDay: number[];
+  totalClicks: number;
+  totalQueries: number;
+}
+interface EngineOverviewValues extends EngineOverviewApiData {
```

Comment:
FYI - this drops/cleans up a good amount of values from ent-search:

- `apiKey` & `apiUrl` - this wasn't being used anywhere, not actually sure where it was coming from (maybe leftover from old logic). `apiKey` should just be grabbed directly from `EngineLogic.values.engine.apiKey` in any case.

- `accountKey` - no longer applies in a SMAS world

- `role` - also not used anywhere AFAICT, should just be pulled directly from `AppLogic.values.myRole`

- `pollingCancelTokenSource` - per a discussion w/ Byron, we will no longer be using/storing axios cancel tokens

- `showDocumentCreationModal` & `creationMode` - I'm going to be spiking out something in a few PRs here where DocumentCreationModal handles its own show/hide state and creation mode state instead of parent views doing it in multiple places.

- `activeApiLog` - Same as the above, I believe ApiLogTables should be in charge of managing the state of its details modal, not the parent. I'm removing it for now on that premise, but if for any reason my belief is incorrect we can come back in later and add it